src/gui/symbology-ng/qgssymbollayerv2widget.cpp

```diff
@@ -110,56 +119,70 @@ void QgsSimpleLineSymbolLayerV2Widget::setSymbolLayer( QgsSymbolLayerV2* layer )
   mLayer = static_cast<QgsSimpleLineSymbolLayerV2*>( layer );
 
   // set units
-  mPenWidthUnitWidget->blockSignals( true );
-  mPenWidthUnitWidget->setUnit( mLayer->widthUnit() );
-  mPenWidthUnitWidget->setMapUnitScale( mLayer->widthMapUnitScale() );
-  mPenWidthUnitWidget->blockSignals( false );
-  mOffsetUnitWidget->blockSignals( true );
-  mOffsetUnitWidget->setUnit( mLayer->offsetUnit() );
-  mOffsetUnitWidget->setMapUnitScale( mLayer->offsetMapUnitScale() );
-  mOffsetUnitWidget->blockSignals( false );
-  mDashPatternUnitWidget->blockSignals( true );
-  mDashPatternUnitWidget->setUnit( mLayer->customDashPatternUnit() );
-  mDashPatternUnitWidget->setMapUnitScale( mLayer->customDashPatternMapUnitScale() );
-  mDashPatternUnitWidget->setMapUnitScale( mLayer->customDashPatternMapUnitScale() );
-  mDashPatternUnitWidget->blockSignals( false );
+  QGS_BLOCK_SIGNAL_DURING_CALL( mPenWidthUnitWidget, setUnit( mLayer->widthUnit() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mPenWidthUnitWidget, setMapUnitScale( mLayer->widthMapUnitScale() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mOffsetUnitWidget, setUnit( mLayer->offsetUnit() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mOffsetUnitWidget, setMapUnitScale( mLayer->offsetMapUnitScale() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mDashPatternUnitWidget, setUnit( mLayer->customDashPatternUnit() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mDashPatternUnitWidget, setMapUnitScale( mLayer->customDashPatternMapUnitScale() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( mDashPatternUnitWidget, setMapUnitScale( mLayer->customDashPatternMapUnitScale() ) )
 
   // set values
-  spinWidth->blockSignals( true );
-  spinWidth->setValue( mLayer->width() );
-  spinWidth->blockSignals( false );
-  btnChangeColor->blockSignals( true );
-  btnChangeColor->setColor( mLayer->color() );
-  btnChangeColor->blockSignals( false );
-  spinOffset->blockSignals( true );
-  spinOffset->setValue( mLayer->offset() );
-  spinOffset->blockSignals( false );
-  cboPenStyle->blockSignals( true );
-  cboJoinStyle->blockSignals( true );
-  cboCapStyle->blockSignals( true );
-  cboPenStyle->setPenStyle( mLayer->penStyle() );
-  cboJoinStyle->setPenJoinStyle( mLayer->penJoinStyle() );
-  cboCapStyle->setPenCapStyle( mLayer->penCapStyle() );
-  cboPenStyle->blockSignals( false );
-  cboJoinStyle->blockSignals( false );
-  cboCapStyle->blockSignals( false );
+  QGS_BLOCK_SIGNAL_DURING_CALL( spinWidth, setValue( mLayer->width() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( btnChangeColor, setColor( mLayer->color() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( spinOffset, setValue( mLayer->offset() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboPenStyle, setPenStyle( mLayer->penStyle() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboJoinStyle, setPenJoinStyle( mLayer->penJoinStyle() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboCapStyle, setPenCapStyle( mLayer->penCapStyle() ) )
```

Comment:
My reasoning was that, since no signal is emitted, it is equivalent.

I understand that if one of the widget has a pointer on one of the others, a signal may be fired by the pointer-to widget. I checked in the custom widgets of qgis and I did not find a widget which holds a pointer on another one.

If you thing my reasoning is wrong, I can restore the old behavior, since it's not directly related to the feature.